src/hotspot/share/prims/jvmtiTagMap.cpp

```diff
@@ -1204,8 +1204,10 @@ void JvmtiTagMap::flush_object_free_events() {
   assert_not_at_safepoint();
   if (env()->is_enabled(JVMTI_EVENT_OBJECT_FREE)) {
     {
+      // If another thread is posting events, let it finish.
+      // This another thread might have safepoints during event callbacks.
+      ThreadBlockInVM tbivm(JavaThread::current());
       MonitorLocker ml(lock(), Mutex::_no_safepoint_check_flag);
```

Comment:
Why not just remove the "no safepoint check" from the MonitorLocker?

Reply:
This is lock is used in different places, including 
`JvmtiTagMap::check_hashmaps_for_heapwalk`
which is called from VMThread. 
Thus JvmtiTagMap_lock can't has rank `safepoint` and this `MonitorLocker` should also has `_no_safepoint_check_flag`.